composer/loggers/mosaicml_logger.py

```diff
@@ -0,0 +1,174 @@
+# Copyright 2021 MosaicML. All Rights Reserved.
+
+from __future__ import annotations
+
+import logging
+import os
+import warnings
+from queue import Queue
+from threading import Thread
+from typing import Optional
+
+import requests
+
+from composer.core.logging import LogLevel, RankZeroLoggerBackend, TLogData
+from composer.core.serializable import Serializable
+from composer.core.types import JSON, Logger, State, StateDict
+
+_MOSAICML_API_KEY_ENV = "MOSAICML_LOGGER_API_KEY"
+_MOSAICML_LOGGER_URL = "https://api.mosaicml.com/v0/log/metric"
+_JOB_ID_ENV = "MOSAICML_LOGGER_JOB_ID"
+_SWEEP_ID_ENV = "MOSAICML_LOGGER_SWEEP_ID"
+
+_STOP_LOGGING_SIGNAL = "STOP_LOGGING"
+
+log = logging.getLogger(__name__)
+
+
+def _send_data(job_id: str, sweep_id: str, data: JSON):
+    response = requests.post(_MOSAICML_LOGGER_URL,
```

Comment:
No need to use `aiohttp` here because this has a 3 second timeout and will not be called that often due to log buffering. Adding the async functionality just introduces unnecessary complexity since it is very unlikely that multiple post requests will interleave.

Reply:
Would it work to move network requests to a background thread? If it's in the main thread, and there's a network issue, then it will crash the training process. Or do we want training to be halted?

Reply:
Right now we catch exceptions + I'll make a change such that the process will exit in the case of a non-transient error